File: app/src/main/java/com/hhy/game/dodgebox/game/Values.java
package com.hhy.game.dodgebox.game;

import android.graphics.Rect;

// values class
// for responsive game
// numbers are found by trial and error method :D
public class Values {
    private int width;
    private int height;

    public Values(int width, int height) {
        this.width = width;
        this.height = height;
    }

    int getWidth() {
        return width;
    }
    int getHeight() {
        return height;
    }
    Rect getBackgroundRect() {
        return new Rect( 0, 0, width, height);
    }
    int getGroundY() {
        return height / 9 * 8;
    }
    Rect getPauseButtonRect() {
        int left = width / 10 * 9;
        int top = height / 50;
        return new Rect(left, top, left + getPausedButtonWidth(), top + getPausedButtonHeight());
    }
    int getPausedButtonWidth() {
        return width / 11;
    }
    int getPausedButtonHeight() {
        return height / 11;
    }
    int getPlayerWidth() {
        return width / 7;
    }
    int getPlayerHeight() {
        return height / 6;
    }
    int getScoreTextSize() {
        return height / 12;
    }
    int getScoreTextX() {
        return width / 100;
    }
    int getScoreTextY() {
        return height / 100;
    }
    int getRecordImgWidth() {
        return width / 32;
    }
    int getRecordImgHeight() {
        return height / 18;
    }
    int getRecordTextSize() {
        return height / 18;
    }
    int getRecordX() {
        return width / 100;
    }
    int getRecordY() {
        return height / 9;
    }
    int getPausedTextSize() {
        return height / 8;
    }
    int getPausedTextX() {
        return width / 10 * 3;
    }
    int getPausedTextY() {
        return height / 7;
    }
    int getBoxWidth() {
        return width / 16;
    }
    int getBoxHeight() {
        return height / 9;
    }
}

File: app/src/main/java/com/hhy/game/dodgebox/game/Values.kt
package com.hhy.game.dodgebox.game

import android.graphics.Rect

// values class
// for responsive game
// numbers are found by trial and error method :D
class Values(val width: Int, val height: Int) {

    val backgroundRect: Rect
        get() = Rect(0, 0, width, height)

    val groundY: Int
        get() = height / 9 * 8

    val pauseButtonRect: Rect
        get() {
            val left = width / 10 * 9
            val top = height / 50
            return Rect(left, top, left + pausedButtonWidth, top + pausedButtonHeight)
        }

    val pausedButtonWidth: Int
        get() = width / 11

    val pausedButtonHeight: Int
        get() = height / 11

    val playerWidth: Int
        get() = width / 7

    val playerHeight: Int
        get() = height / 6

    val scoreTextSize: Int
        get() = height / 12

    val scoreTextX: Int
        get() = width / 100

    val scoreTextY: Int
        get() = height / 100

    val recordImgWidth: Int
        get() = width / 32

    val recordImgHeight: Int
        get() = height / 18

    val recordTextSize: Int
        get() = height / 18

    val recordX: Int
        get() = width / 100

    val recordY: Int
        get() = height / 9

    val pausedTextSize: Int
        get() = height / 8

    val pausedTextX: Int
        get() = width / 10 * 3

    val pausedTextY: Int
        get() = height / 7

    val boxWidth: Int
        get() = width / 16

    val boxHeight: Int
        get() = height / 9

}